### PR TITLE
update features

### PR DESCRIPTION
**Pull request by lijialin03**
### PR types

Others
### PR changes

Others
### Describe

1. specifies the input parameter type of transformer
2. add input_dim parameter of mlp
3. overwrite __str__ of sup constraints and validator
4. make label_keys of mat_dataset be optional
5. change judgment of weight_dict in all kinds of loss

### CODE REVIEW

File: ppsci/arch/base.py [hunk 1]
```diff
@@ -80,10 +81,26 @@ def split_to_dict(
...
 
-    def register_input_transform(self, transform):
+    def register_input_transform(
+        self,
+        transform: Callable[[Dict[str, paddle.Tensor]], Dict[str, paddle.Tensor]],
+    ):
+        """Register input transform.
+
+        Args:
+            transform (Callable[[Dict[str, paddle.Tensor]], Dict[str, paddle.Tensor]]): Transform, parameter and return of it should be one dict.
```
Comment (HydrogenSulfate) [inline ppsci/arch/base.py:91]: Input transform of network, receive a single tensor dict and return a single tensor dict.

File: ppsci/arch/base.py [hunk 2]
```diff
@@ -80,10 +81,26 @@ def split_to_dict(
...
 
-    def register_output_transform(self, transform):
+    def register_output_transform(
+        self,
+        transform: Callable[[Dict[str, paddle.Tensor]], Dict[str, paddle.Tensor]],
+    ):
+        """Register output transform.
+
+        Args:
+            transform (Callable[[Dict[str, paddle.Tensor]], Dict[str, paddle.Tensor]]): Transform, parameter and return of it should be one dict.
```
Comment (HydrogenSulfate) [inline ppsci/arch/base.py:102]: Output transform of network, receive a single tensor dict and return a single tensor dict.

File: ppsci/arch/mlp.py [hunk 3]
```diff
@@ -33,6 +33,7 @@ class MLP(base.Arch):
         activation (str, optional): Name of activation function. Defaults to "tanh".
         skip_connection (bool, optional): Whether to use skip connection. Defaults to False.
         weight_norm (bool, optional): Whether to apply weight norm on parameter(s). Defaults to False.
+        input_dim (int, optional): Number of input's dimension.
```
Comment (HydrogenSulfate) [inline ppsci/arch/mlp.py:36]: 1. 改为 `Optional[int] = None`
2. 参数说明默认为加上 `Defaults to None.`

File: ppsci/arch/mlp.py [hunk 4]
```diff
@@ -48,6 +49,7 @@ def __init__(
         activation: str = "tanh",
         skip_connection: bool = False,
         weight_norm: bool = False,
+        input_dim: int = None,
```
Comment (HydrogenSulfate) [inline ppsci/arch/mlp.py:52]: Optional[int] = None

File: ppsci/data/dataset/mat_dataset.py [hunk 5]
```diff
@@ -34,7 +34,7 @@ class MatDataset(io.Dataset):
     Args:
         file_path (str): Mat file path.
         input_keys (Tuple[str, ...]): List of input keys.
-        label_keys (Tuple[str, ...]): List of label keys.
+        label_keys (Optional[Tuple[str, ...]]): List of label keys. Defaults to empty.
```
Comment (HydrogenSulfate) [inline ppsci/data/dataset/mat_dataset.py:37]: 默认值非None的参数不要用typing.Optional，
``` py
label_keys (Tuple[str, ...], optional): List of label keys. Defaults to ().
```

File: ppsci/data/dataset/mat_dataset.py [hunk 6]
```diff
@@ -57,7 +57,7 @@ def __init__(
         self,
         file_path: str,
         input_keys: Tuple[str, ...],
-        label_keys: Tuple[str, ...],
+        label_keys: Optional[Tuple[str, ...]] = (),
```
Comment (HydrogenSulfate) [inline ppsci/data/dataset/mat_dataset.py:60]: `label_keys: Tuple[str, ...] = (),`

File: ppsci/data/dataset/mat_dataset.py [hunk 7]
```diff
@@ -153,7 +153,7 @@ class IterableMatDataset(io.IterableDataset):
     Args:
         file_path (str): Mat file path.
         input_keys (Tuple[str, ...]): List of input keys.
-        label_keys (Tuple[str, ...]): List of label keys.
+        label_keys (Optional[Tuple[str, ...]]): List of label keys. Defaults to empty.
```
Comment (HydrogenSulfate) [inline ppsci/data/dataset/mat_dataset.py:156]: 修改同上

File: ppsci/data/dataset/mat_dataset.py [hunk 8]
```diff
@@ -176,7 +176,7 @@ def __init__(
         self,
         file_path: str,
         input_keys: Tuple[str, ...],
-        label_keys: Tuple[str, ...],
+        label_keys: Optional[Tuple[str, ...]] = (),
```
Comment (HydrogenSulfate) [inline ppsci/data/dataset/mat_dataset.py:179]: 修改同上

**Review by HydrogenSulfate — CHANGES_REQUESTED** [review]

File: ppsci/arch/mlp.py (hunk 3 above)
Comment (lijialin03) [reply ppsci/arch/mlp.py]: 改为 input_dim (Optional[int], optional) 了

**Review by HydrogenSulfate — APPROVED** [review]
LGTM

**Pull request merged**